hyperspy/io_plugins/jeol.py

```diff
@@ -0,0 +1,649 @@
+                                "Doc"
+                            ]["RealTime"],
+                        },
+                    },
+                },
+            },
+            "General": {
+                "original_filename": os.path.basename(filename),
+                "date": datefile.date().isoformat(),
+                "time": datefile.time().isoformat(),
+                "title": "EDX",
+            },
+            "Signal": {
+                "record_by": "spectrum",
+                "quantity": "X-rays (Counts)",
+                "signal_type": "EDS_" + mode,
+            },
+        }
+
+        dictionary = {
+            "data": data,
+            "axes": axes,
+            "metadata": metadata,
+            "original_metadata": header,
+        }
+    else:
+        print("Not a valid JEOL pts format")
+
+    fd.close()
+
+    return dictionary
+
+
+def parsejeol(fd):
+    final_dict = {}
+    tmp_list = []
+    tmp_dict = final_dict
+
+    mark = 1
+    while abs(mark) == 1:
+        mark = np.fromfile(fd, "b", 1)[0]
+        if mark == 1:
+            str_len = np.fromfile(fd, "<i", 1)[0]
+            kwrd = fd.read(str_len).rstrip(b"\x00")
+
+            if (
+                kwrd == b"\xce\xdf\xb0\xc4"
+            ):  # correct variable name which might be 'Port'
+                kwrd = "Port"
+            elif (
+                kwrd[-1] == 222
+            ):  # remove undecodable byte at the end of first ScanSize variable
+                kwrd = kwrd[:-1].decode("utf-8")
+            else:
+                kwrd = kwrd.decode("utf-8")
+            val_type, val_len = np.fromfile(fd, "<i", 2)
+            tmp_list.append(kwrd)
+            if val_type == 0:
+                tmp_dict[kwrd] = {}
+            else:
+                c_type = jTYPE[val_type]
+                arr_len = val_len // np.dtype(c_type).itemsize
+                if c_type == "c":
+                    value = fd.read(val_len).rstrip(b"\x00")
+                    value = value.decode("utf-8").split("\x00")
+                    # value = os.path.normpath(value.replace('\\','/')).split('\x00')
+                else:
+                    value = np.fromfile(fd, c_type, arr_len)
+                if len(value) == 1:
+                    value = value[0]
+                if kwrd[-5:-1] == "PAGE":
+                    kwrd = kwrd + "_" + value
+                    tmp_dict[kwrd] = {}
+                    tmp_list[-1] = kwrd
+                elif kwrd == "CountRate" or kwrd == "DeadTime":
+                    tmp_dict[kwrd] = {}
+                    tmp_dict[kwrd]["value"] = value
+                elif kwrd == "Limits":
+                    print("skip")
+                elif val_type == 14:
+                    tmp_dict[kwrd] = {}
+                    tmp_dict[kwrd]["index"] = value
+                else:
+                    tmp_dict[kwrd] = value
+            if kwrd == "Limits":
+                print("skip2")
```

Comment:
Same as above.

Reply:
Same as above but in int32 instaed of float32
- first 16 bytes are encode in int32 and looks like limit values (10, 1, 100000000, 1)
- next 4 bytes are ascii character and looks like number format (%d)
- next 12 bytes are unclear for me
- next 4 bytes are ascii character and are units (mag)
- lest 12 byes are again unclear for me